src/frontend/web_application/src/scenes/UserSecurity/components/OpenPGPKey/index.tsx

```diff
@@ -0,0 +1,177 @@
+}
+function OpenPGPKey({
+  className,
+  publicKeyArmored,
+  privateKeyArmored,
+  onDeleteKey,
+  i18n,
+  editMode = false,
+}: Props) {
+  const [keyDetails, setKeyDetails] = React.useState<KeyDetails>();
+  const [loading, setIsLoading] = React.useState<boolean>(false);
+  const locale = useSelector<RootState, string>((state) => state.i18n.locale);
+
+  React.useEffect(() => {
+    setIsLoading(true);
+    getPGPManager().then(async ({ getKeyFromASCII }) => {
+      const pubKey = await getKeyFromASCII(publicKeyArmored);
+
+      pubKey && setKeyDetails(await getKeyDetails(pubKey));
+      setIsLoading(false);
+    });
+  }, [publicKeyArmored]);
+
+  const privateKeyDataUrl = `data:application/x-pgp;base64,${strToBase64(
+    privateKeyArmored
+  )}`;
+
+  const handleDeleteKey = () => {
+    if (!keyDetails) {
+      return;
+    }
+
+    onDeleteKey(keyDetails.fingerprint);
+  };
+
+  const openpgpStatuses = {
+    invalid: i18n._('openpgp.status.invalid', undefined, {
+      defaults: 'Invalid',
+    }),
+    expired: i18n._('openpgp.status.expired', undefined, {
+      defaults: 'Expired',
+    }),
+    revoked: i18n._('openpgp.status.revoked', undefined, {
+      defaults: 'Revoked',
+    }),
+    valid: i18n._('openpgp.status.valid', undefined, { defaults: 'Valid' }),
+    no_self_cert: i18n._('openpgp.status.no_self_cert', undefined, {
+      defaults: 'No self cert',
+    }),
+  };
+
+  return (
+    <div className={classnames(['m-openpgp-key', className])}>
+      <div className="m-openpgp-key__main">
+        <Spinner isLoading={loading} />
+        {keyDetails && (
+          <>
+            <div className="m-openpgp-key__icon">
+              {/* @ts-ignore */}
+              <Icon type="key" />
+            </div>
+            <div className="m-openpgp-key__fingerprint">
+              {keyDetails.fingerprint.toUpperCase()}
+            </div>
+
+            <div className="m-openpgp-key__actions">
+              <Link
+                button
+                plain
+                href={privateKeyDataUrl}
+                download="private-key.asc"
+              >
+                <Trans id="openpgp-key.download">
+                  Save and keep in a safe place.
+                </Trans>
+                {' '}
+                {/* @ts-ignore */}
+                <Icon type="download" />
+              </Link>
+              {editMode && (
+                // @ts-ignore
+                <Button color="alert" onClick={handleDeleteKey}>
+                  {/* @ts-ignore */}
+                  <Icon type="remove" />
+                  <span className="show-for-sr">
+                    <Trans id="openpgp.action.remove-key">Remove</Trans>
+                  </span>
+                </Button>
+              )}
+            </div>
+          </>
+        )}
+      </div>
+
+      {keyDetails && (
+        <div className="m-openpgp-key__summary">
+          <span>{keyDetails?.userIds}</span>
```

Comment:
`userIds` are thrown as is, without separators, which is difficult to read. We may want to dive in a little bit to see how to display this in a better way.

Reply:
fixed